### ISSUE TITLE
How to choose warm-up steps depending on the train steps during pre-training?

### ISSUE THREAD

**Issue by nellymin** [google-research/bert#369]
In the paper it says:
> We train with batch size of 256 sequences (256 sequences * 512 tokens = 128,000 tokens/batch) for 1,000,000 steps, which is approximately 40 epochs over the 3.3 billion word corpus. We use Adam with learning rate of 1e-4, β1 = 0.9, β2 = 0.999, L2 weight decay of 0.01, learning rate warm-up over the first 10,000 steps, and linear decay of the learning rate  
That is 1% of the overall train steps or another way of looking at it is less than half an epoch. 

Do you have any advice on choosing a good number for pre-training?

**Comment by mahdirezaey** [comment]
maybe 
https://github.com/google-research/bert/issues/649

can help
